### DISCUSSION TITLE
When using UUID for User ID, can't delete permissions.

### DISCUSSION THREAD

**Discussion by gabrielgasp** [spatie/laravel-permission#2365] [Q&A] [answered]
PHP: 8.2.0
Laravel: 10.0
Spatie/Laravel-Permission: 5.9.1

My users table use UUID for primary key. Followed the instructions here to [here](https://spatie.be/docs/laravel-permission/v5/advanced-usage/uuid#content-migrations) to adapt the package so that roles and permissions worked with it.

I'm only changing the "model_morph_key" to UUID, primary keys of tables created by the package are left as is.

When I try to delete a permission, the application is trying to do the following query: `delete from "model_has_permissions" where "model_has_permissions"."model_id" = 1 and "model_type" = Spatie\Permission\Models\Permission`.

This results in an error because the permission_id (1 in the query) is not a valid uuid. I'm not sure if I did something wrong but I don't see why the package would try to query the `"model_has_permissions"."model_id"` using the permission id when there is a `permission_id` column in the table.

Edit: This problem doesn't happen when trying to delete a role, which has pretty much the same database/migration structure.

**Comment by erikn69** [comment, marked as answer]
Hi @GabrielGS447, i make a PR #2366 to fix this, please review it

**Reply by gabrielgasp** [reply]
I'm glad that you found the problem and implemented a solution. That said, I don't believe I'm supposed to review/approve the changes you made as I'm not familiar with the internal code of the package. I believe it should be reviewed by another contributor of the project.

For the time being I circumvented the problem by also changes the Permissions and Roles tables Primary Keys to UUID.

**Reply by gabrielgasp** [reply]
Did as you asked and it solved the problem. Approved the PR.

**Reply by drbyte** [reply]
Released in v`5.10.0`
